tests/CMake/SiloTestFunctions.cmake

```diff
@@ -0,0 +1,197 @@
+# herein do not necessarily state  or reflect those of the United States
+# Government or Lawrence Livermore National Security, LLC, and shall not
+# be used for advertising or product endorsement purposes.
+#
+##############################################################################
+
+
+# NAME: name of test, required
+# LANG: language of source files (optional)
+#       e.g. for CXX source files with '.C' extension
+# SRC:  list of source files, required
+function(silo_add_test)
+    set(options)
+    set(oneValueArgs NAME LANG)
+    set(multiValueArgs SRC)
+    cmake_parse_arguments(sat "${options}" "${oneValueArgs}"
+        "${multiValueArgs}" ${ARGN} )
+
+    if("NAME" IN_LIST sat_KEYWORDS_MISSING_VALUES)
+        message(WARNING "silo_add_test: NAME argument is missing a value.")
+        return()
+    elseif(NOT DEFINED sat_NAME)
+        message(WARNING "silo_add_test requires NAME argument.")
+        return()
+    endif()
+    if("SRC" IN_LIST sat_KEYWORDS_MISSING_VALUES)
+        message(WARNING "silo_add_test: SRC argument is missing a value.")
+        return()
+    elseif(NOT DEFINED sat_SRC)
+        message(WARNING "silo_add_test: requires SRC argument")
+        return()
+    endif()
+
+
+    add_executable(${sat_NAME} ${sat_SRC})
+    if(WIN32)
+        # change output directory using generator expression to avoid
+        # VS adding per-configuration subdir
+        set_target_properties(${sat_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY
+            ${silo_test_output_dir}/$<$<CONFIG:Debug>:>)
+    endif()
+    if("LANG" IN_LIST sat_KEYWORDS_MISSING_VALUES)
+        message(WARNING "silo_add_test: LANG argument is missing a value.")
+        return()
+    elseif(DEFINED sat_LANG)
+        set_property(SOURCE ${sat_SRC} PROPERTY
+                      LANGUAGE ${sat_LANG})
+    endif()
+
+    add_dependencies(${sat_NAME} silo)
+    target_compile_definitions(${sat_NAME} PRIVATE PDB_LITE) 
+
+    target_link_libraries(${sat_NAME} $<TARGET_LINKER_FILE:silo>)
+
+    target_include_directories(${sat_NAME}  PRIVATE
+        ${silo_build_include_dir}
+        ${Silo_SOURCE_DIR}/src/silo
+        ${SILO_TESTS_SOURCE_DIR})
+
+    set_target_properties(${sat_NAME} PROPERTIES FOLDER testing/tests)
+
+    if(SILO_ENABLE_HDF5 AND (TARGET hdf5 OR HDF5_FOUND))
+        if(TARGET hddf5)
+            target_link_libraries(${sat_NAME} hdf5)
+        else()
+            target_include_directories(${sat_NAME} PRIVATE ${HDF5_INCLUDE_DIRS})
+            target_link_libraries(${sat_NAME} ${HDF5_LIBRARIES})
+        endif()
+    endif()
+
+    if(WIN32)
+        target_compile_definitions(${sat_NAME} PRIVATE PDB_LITE)
+    endif()
+endfunction()
+
+# this function adds individual test-runner targets 
+# allows individual tests to be run via 'make run_testall_hdf5' 
+# 'run_ is prefixed to runner name, to prevent target name collisions with
+# test executables
+function(silo_add_test_runner)
```

Comment:
Is there a way to associate an expected (textual or binary silo file) output with a test to compare agains? I think that may also be CTest related functionality and if so, no worry to ignore here.

Reply:
The silo_add_test_runner function was part of my previous implementation that created 'run_<testname>_<testargs>' targets that could be run from the build. Klunky. Weird names. I forgot to remove that function when I switched to creation of the 'check' target. I will remove it to avoid confusion.